### PR TITLE
[configure] Guard debug code under `-debug` flag

### LINKED ISSUE #14888: Spurious new lines in the output of configure.
#### Description of the problem

I ran `./configure -profile devel` and got the following lines in the output:

```
2.8.2 / 282            
Warning: You are using Dune < 2.9, the install procedure will not respect the -docdir and -configdir configure directives; please see dev/doc/INSTALL.make.md for more information
4.09.1 / 4091
You have OCaml 4.09.1. Good!
1.8.1 / 181
You have OCamlfind 1.8.1. Good!
You have native-code compilation. Good!
1.11 / 1110
You have the Zarith library 1.11 installed. Good!
3.1.1 / 311
```

I believe that all the lines of the form `XXX / XXX` are debug lines that weren't meant to be merged into master. Is that a correct interpretation @ejgallego?

#### Coq Version

cd8f5031c0915ca7d57287374dc36fde7d820a67

### PR DESCRIPTION
**Pull request by ejgallego**
Fixes #14888

### CODE REVIEW

File: tools/configure/util.ml [hunk 1]
```diff
@@ -144,9 +144,12 @@ let numeric_prefix_list s =
   | [v1;v2;""] -> [v1;v2;"0"] (* e.g. because it ends with ".beta" *)
   | v -> v
 
+let debug_version_detection = false
```
Comment (SkySkimmer) [inline tools/configure/util.ml:147]: We should either actually remove the code or make this controllable from the command line imo

Comment (Alizter) [reply tools/configure/util.ml:147]: I'm for completely removing it, if it's really needed to be kept around we can always fish it out of the git history.

Comment (ejgallego) [reply tools/configure/util.ml:147]: It is actually usual for these kind of system APIs to actually have debug parameters, and keep the unexposed to the final user.

Comment (Zimmi48) [reply tools/configure/util.ml:147]: I'm also in favor of completely removing it.

File: tools/configure/cmdArgs.ml [hunk 2]
```diff
@@ -182,7 +184,8 @@ let args_options = Arg.align [
     " URL of the coq website";
   "-warn-error", arg_bool (fun p warn_error -> { p with warn_error }),
     "(yes|no) Make OCaml warnings into errors (default no)";
-  "-profile", arg_profile, Profiles.doc
+  "-profile", arg_profile, Profiles.doc;
+  "-debug", arg_set (fun p -> { p with debug = true }), "enable debug information for package detection"
```
Comment (Alizter) [inline tools/configure/cmdArgs.ml:188]:
```suggestion
  "-debug", arg_set (fun p -> { p with debug = true }), "Enable debug information for package detection"
```

**Review by Alizter — COMMENTED** [review]
Sure we can keep a `-debug` flag around for `./configure`. I am just a bit weary of the future, if we need to debug other things do we also use this flag?

Perhaps it would be better to call it `-debug-package-detection` so we don't have to do anything silly in the future.

File: tools/configure/cmdArgs.ml (hunk 2 above)
Comment (SkySkimmer) [reply tools/configure/cmdArgs.ml:188]: Waiting for this (or just remove the code, I don't think it's actually useful)

**Comment by ejgallego** [comment]
I guess we can merge this.

**Comment by Alizter** [comment]
I will merge tomorrow.

**Comment by Alizter** [comment]
We want to target 8.14 right? @silene do you agree?

**Comment by Zimmi48** [comment]
Yes, that's 8.15 material.

**Comment by Alizter** [comment]
@coqbot merge now

**Review by Alizter — APPROVED** [review]

**Comment by Alizter** [comment]
@coqbot merge now

**Pull request merged**